### PR TITLE
Fixed nil pointer when no Jaeger is suitable for sidecar injection

### PR DESCRIPTION
**Pull request by jpkrohling**
Signed-off-by: Juraci Paixão Kröhling <juraci@kroehling.de>

### CODE REVIEW

**Comment by jpkrohling** [comment]
Could you please review, @rubenvp8510 ?

File: pkg/autodetect/main.go [hunk 1]
```diff
@@ -99,7 +101,15 @@ func (b *Background) requireUpdates(deps *appsv1.DeploymentList) []*appsv1.Deplo
 		} else {
 			// Try to update the sidecar if is required
 			jaeger := inject.Select(dep, instances)
-			updated := inject.UpdateSideCar(jaeger, dep)
+			if jaeger == nil {
+				log.WithFields(log.Fields{
+					"deploymentName":      dep.Name,
+					"deploymentNamespace": dep.Namespace,
+				}).Debug("no suitable jaeger for this instance, skipping injection")
+				return nil
```
Comment (rubenvp8510) [inline pkg/autodetect/main.go]: I think this should be a continue not return, we need to process the rest of the deployments.

Comment (jpkrohling) [reply pkg/autodetect/main.go]: Changed.

**Comment by rubenvp8510** [comment]
LGTM

**Pull request merged**